examples/llama.py

```diff
@@ -241,21 +241,12 @@ def sample(logits, temperature):
       # if the weight is copied across models, it's simple
       # TODO: assert they are the same
       if w0.shape == mv.shape:
-        mv.lazydata.realized = w0
-        w0._buf = None
+        mv.assign(w0)
+        mv.realize()
         continue
 
-      # we have to concatenate them, create tensors
-      w0t = Tensor.empty(*w0.shape)
-      w1t = Tensor.empty(*w1.shape)
-      w0t.lazydata.realized = w0
-      w1t.lazydata.realized = w1
-
-      # terrible hacks. force create the output buffer as float16
-      mv.lazydata.realized = Device._buffers[Device.DEFAULT].empty(mv.shape, dtype=w0.dtype)
-
-      if w0.shape[0] != mv.shape[0]: mv.assign(w0t.cat(w1t, dim=0))
-      elif w0.shape[1] != mv.shape[1]: mv.assign(w0t.cat(w1t, dim=1))
+      if w0.shape[0] != mv.shape[0]: mv.assign(w0.cat(w1, dim=0))
+      elif w0.shape[1] != mv.shape[1]: mv.assign(w0.cat(w1, dim=1))
```

Comment:
btw you need to put hacks back it, currently using a lot of ram, i'm not sure how to fix them

Reply:
I guess it's easier to just use this [merge-meights.py](https://github.com/randaller/llama-chat/blob/main/merge-weights.py) or port it on tinygrad - that will make merged file with weights and you can load it as 7B without any copies in memory.